electrumsv/wallet_database/functions.py

```diff
@@ -768,22 +768,25 @@ def read_payment_requests(db: sqlite3.Connection, account_id: int,
         for t in db.execute(sql, sql_values).fetchall() ]
 
 
-# TODO(1.4.0) Remove when we have replaced with a reference server equivalent.
-# @replace_db_context_with_connection
-# def read_reorged_transactions(db: sqlite3.Connection, reorg_height: int) -> List[bytes]:
-#     """
-#     Identify all transactions that were verified in the orphaned chain as part of a reorg.
-#     """
-#     sql = (
-#         "SELECT tx_hash "
-#         "FROM Transactions "
-#         f"WHERE block_height>? AND flags&{TxFlags.STATE_SETTLED}!=0"
-#     )
-#     sql_values = (reorg_height,)
-#     cursor = db.execute(sql, sql_values)
-#     rows = [ tx_hash for (tx_hash,) in cursor.fetchall() ]
-#     cursor.close()
-#     return rows
+@replace_db_context_with_connection
+def read_reorged_transactions(db: sqlite3.Connection,
+        orphaned_block_hashes: List[bytes]) -> List[bytes]:
+    """
+    Identify all transactions that were verified in the orphaned chain as part of a reorg.
+    """
+    # TODO(1.4.0) Update unittest
+
+    sql = f"""
+        SELECT tx_hash
+        FROM Transactions
+        WHERE block_hash IN ({','.join('?' for k in orphaned_block_hashes)})
+        AND flags&{TxFlags.STATE_SETTLED}!=0
```

Comment:
`flags&{TxFlags.MASK_STATE}={TxFlags.STATE_SETTLED}`. In the longer run we will be likely transitioning to the state values being a numerical sequence not bit flags, and this will work for that.